### PR TITLE
replace boring items with fun items on spawn

### PR DESCRIPTION
**Pull request by Slava0135**
## Media

Peek 2024-03-13 20-12

## Breaking changes

**Changelog**

:cl:
- add: added fun!

### CODE REVIEW

**Comment by Slava0135** [comment]
also I made it so if you eat banana there is chance it spawns explosive banana peel

**Comment by Slava0135** [comment]
fixed containers (now if you eat banana, explosive banana peel appears in hand rather than ground)

**Comment by Dragonjspider** [comment]
Obligitory "is this for april fools" question (I assume it is but funnier if not :trollface:  )

**Comment by Slava0135** [comment]
i added more fun items to be replaced (ideas are welcome)

**Comment by Slava0135** [comment]
me when there is a chance for nitrous gas tank to be replaced with plasma tank :trollface:

File: Content.Server/ReplaceOnSpawn/ReplaceOnSpawnSystem.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        // because OnMapInit is called when entity is spawned it can be put in container later which breaks stuff
+        // so have to do it this way instead
+        var toReplace = new ValueList<(EntityUid, ReplaceOnSpawnComponent)>();
+        var query = EntityQueryEnumerator<ReplaceOnSpawnComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            if (!_random.Prob(comp.Chance))
+            {
+                RemComp<ReplaceOnSpawnComponent>(uid);
```
Comment (metalgearsloth) [inline Content.Server/ReplaceOnSpawn/ReplaceOnSpawnSystem.cs:26]:
```suggestion
            {
                RemCompDeferred<ReplaceOnSpawnComponent>(uid);
```

Modifying while enumerating can break stuff.

**Pull request merged**